### PR TITLE
FEATURE: Center all dialogs vertically

### PR DESCRIPTION
**Pull request by gerhard-boden**
Center all dialogs vertically with flexbox. 
Dialogs using the old markup will maintain the old positioning as fallback.

### CODE REVIEW

**Comment by johannessteu** [comment]
👍  Tested on Mac in
- Chrome  49.0.2623.110 (64-bit)
- Firefox 44.0.2 & 45.0.1
- Safari 9.1

**Comment by gerhard-boden** [comment]
Thanks for the first review @johannessteu!

Since we are going with flexbox now, i will close the `table-cell` PR in favour of this one.

Furthermore the node insert modal will have a fixed offset from the top, to prevent the dialog from jumping when folding/unfolding sections. 

All other dialogs will be dead center from now on.

*WARNING* Custom backend modules using the `neos-modal` will need the new markup too! Basically it's just another wrapper for the content. Schould we mark this as breaking?

**Comment by johannessteu** [comment]
Yes, Breaking would be good. The modal is a very common class to use for custom modules. I used the modal box a lot to pop up before something is deleted

**Comment by aertmann** [comment]
Thanks for taking care, too bad it cannot be done without changing the markup. I think we need to make it backwards compatible, since it's quite breaking. A simple way to do that would be to keep the existing behavior and introduce the new behavior with a new class name `.neos-modal-centered` or similar. This also allows for allowing both ways and not specifically for the insert new panel.

**Comment by kdambekalns** [comment]
Yes, please make it BC, otherwise it should not be merged until we are about to do the next major release… I like the suggestion from Aske.

**Comment by gerhard-boden** [comment]
Thanks for the feedback everybody! I rewrote this so the neos backend will take advantage of flexbox, while modals using the classic markup will show up as they used to (so custom backend module dialogs will look just the same). The new markup can be applied easily though.

@johannessteu maybe you can also check one of your custom backend modules if the dialogs show up like they used to.

File: TYPO3.Neos/Resources/Private/Styles/Content/_InsertNodePanel.scss [hunk 1]
```diff
@@ -1,4 +1,6 @@
 .neos-insert-node-panel {
+	@include flex-align(flex-start);
+	top: 140px;
```
Comment (aertmann) [inline TYPO3.Neos/Resources/Private/Styles/Content/_InsertNodePanel.scss]: this shouldn't be needed right?

Comment (aertmann) [inline TYPO3.Neos/Resources/Private/Styles/Content/_InsertNodePanel.scss]: nvm, it is

**Comment by aertmann** [comment]
a bit more was needed to make it truly backwards compatible

@gerhard-boden: pushed a PR to your branch https://github.com/gerhard-boden/neos-development-collection/pull/4

**Comment by gerhard-boden** [comment]
Hi @aertmann. Thanks for your PR but I don't fully understand what you mean.

Took a quick look at it but what part makes it more backards compatible? 
Maybe we don't mean the same. When I said "backards compatible" I was talking about custom backend modules that use the "old" modal markup. They will still work after this CSS changes (at least they did when I tested them). 

The node panel is not backwards compatible because I doubt it has to be (because nobody will copy the current markup and use it for something else)

**Comment by johannessteu** [comment]
Hey @gerhard-boden 

the modal is not really centered for me. Could also be a bit bigger on big screens?!

Before:
screen shot 2016-04-14 at 09 15 10

After:
screen shot 2016-04-14 at 09 15 57

**Comment by gerhard-boden** [comment]
@johannessteu thanks for your feedback!

All dialogs are centered except for the node insert dialog. The reason being this:
https://github.com/neos/neos-development-collection/pull/434#issuecomment-207437072
(jumping buttons)

I can't reproduce your problem with the dialog width. Did you run `grunt rebuild`?

Here's what the dialog looks like with this PR:
image

Here's the current master:
image

**Comment by johannessteu** [comment]
Ah, `grunt compile` did the trick. Thought everything compiled is included. So 👍

**Pull request merged**